### PR TITLE
#143401485 Add front end tests.

### PR DESCRIPTION
**Pull request by amutava**
#### Test add check, test pushover, test my checks and test verify email
#### After cloning the repo:

- create a virtual environment 
- cd webapps/healthchecks-boomslang
- run python manage.py test hc.front.tests 
#### Relevant pivotal tracker story
Id : 143401485
Link : https://www.pivotaltracker.com/story/show/143401485

### CODE REVIEW

File: hc/front/tests/test_add_check.py [hunk 1]
```diff
@@ -7,14 +7,20 @@ class AddCheckTestCase(BaseTestCase):
...
         assert Check.objects.count() == 1
 
-    ### Test that team access works
+    # ##Test that team access works
+    def test_team_access_works(self):
+        url = "/checks/add/"
+        self.client.login(username="bob@example.org", password="password")
+        resp = self.client.post(url)
+        self.assertRedirects(resp, "/checks/")
+        assert Check.objects.count() == 1
```
Comment (collinmutembei) [inline hc/front/tests/test_add_check.py]: Have a look at Django's documentation on assert statements [here](https://goo.gl/Ra6s5u)

File: hc/front/tests/test_add_pushbover.py [hunk 2]
```diff
@@ -18,8 +18,8 @@ def test_it_adds_channel(self):
         session.save()
 
         params = "pushover_user_key=a&nonce=n&prio=0"
-        r = self.client.get("/integrations/add_pushover/?%s" % params)
-        assert r.status_code == 302
+        resp = self.client.get("/integrations/add_pushover/?%s" % params)
+        assert resp.status_code == 302
```
Comment (collinmutembei) [inline hc/front/tests/test_add_pushbover.py]: 👀

File: hc/front/tests/test_add_pushbover.py [hunk 3]
```diff
@@ -28,10 +28,19 @@ def test_it_adds_channel(self):
...
     ### Test that pushover validates priority
+    def test_push_over_validates_priority(self):
+        self.client.login(username="alice@example.org", password="password")
+        session = self.client.session
+        session["po_nonce"] = "n"
+        session.save()
+
+        params = "pushover_user_key=a&nonce=n&prio=8"
+        resp = self.client.get("/integrations/add_pushover/?%s" % params)
+        assert resp.status_code == 400
```
Comment (collinmutembei) [inline hc/front/tests/test_add_pushbover.py]: 👀

File: hc/settings.py [hunk 4]
```diff
@@ -17,6 +17,13 @@
 DATABASES = {}
 
 DATABASES['default'] = dj_database_url.config(conn_max_age=600)
+DATABASES['default']['ENGINE'] = 'django.db.backends.postgresql_psycopg2'
+# DATABASES = {
```
Comment (collinmutembei) [inline hc/settings.py]: Why is this statement commented out? Is it still needed?

**Review by collinmutembei — CHANGES_REQUESTED** [review]
@andela-amutava @valeria-chemtai Attend to the comments below and ensure that the tests are passing.

File: hc/front/tests/test_add_check.py (hunk 1 above)
Comment (valeria-chemtai) [reply hc/front/tests/test_add_check.py]: @collinmutembei the issues highlighted have been rectified.
Kindly check.

File: hc/api/tests/test_list_checks.py [hunk 5]
```diff
@@ -33,13 +33,15 @@ def get(self):
...
+        resp = self.get()
         ### Assert the response status code
+        self.assertEqual(resp.status_code, 200)
 
-        doc = r.json()
+        doc = resp.json()
         self.assertTrue("checks" in doc)
 
         checks = {check["name"]: check for check in doc["checks"]}
+        import ipdb; ipdb.set_trace()
```
Comment (collinmutembei) [inline hc/api/tests/test_list_checks.py]: Remember to remove debug statements before pushing code.

Comment (valeria-chemtai) [reply hc/api/tests/test_list_checks.py]: @collinmutembei we must have committed that by mistake since we were doing front end tests

Comment (egichuri) [reply hc/api/tests/test_list_checks.py]: @valeria-chemtai seems like you might still need to delete this since it's on your branch.

Comment (valeria-chemtai) [reply hc/api/tests/test_list_checks.py]: @andela-egichuri, on it.

File: hc/settings.py [hunk 6]
```diff
@@ -17,6 +17,8 @@
 DATABASES = {}
 
 DATABASES['default'] = dj_database_url.config(conn_max_age=600)
+DATABASES['default']['ENGINE'] = 'django.db.backends.postgresql_psycopg2'
```
Comment (egichuri) [inline hc/settings.py:20]: Do we need this?

**Comment by egichuri** [comment]
Slight merge conflict here. We'll use this to explain how to solve merge conflicts.

**Pull request merged**